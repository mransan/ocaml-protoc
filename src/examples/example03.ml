let () = 
  let v = Example03_types.Some "This is an example" in  
  Format.fprintf Format.std_formatter "%a\n" Example03_pp.pp_string_some v
