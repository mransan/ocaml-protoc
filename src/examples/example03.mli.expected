
(** Code for example03.proto *)

(* generated from "example03.proto", do not edit *)



(** {2 Types} *)

type string_some_none = unit

type string_some_t =
  | None
  | Some of string

and string_some = {
  t : string_some_t option;
}


(** {2 Basic values} *)

val default_string_some_none : unit
(** [default_string_some_none ()] is the default value for type [string_some_none] *)

val default_string_some_t : unit -> string_some_t
(** [default_string_some_t ()] is the default value for type [string_some_t] *)

val default_string_some : 
  ?t:string_some_t option ->
  unit ->
  string_some
(** [default_string_some ()] is the default value for type [string_some] *)


(** {2 Formatters} *)

val pp_string_some_none : Format.formatter -> string_some_none -> unit 
(** [pp_string_some_none v] formats v *)

val pp_string_some_t : Format.formatter -> string_some_t -> unit 
(** [pp_string_some_t v] formats v *)

val pp_string_some : Format.formatter -> string_some -> unit 
(** [pp_string_some v] formats v *)
